poems_parser.py

```diff
@@ -8,86 +8,136 @@
     for row in table.find_all('td', valign='top'):
         poem_name = row.find('a', class_='linki2').text
-        poem_ur = url + row.find('a', class_='linki2')['href']
+        poem_url = row.find('a', class_='linki2')['href']
         try:
             poem_year = row.find('span').text
-        except AttributeError:
-            continue
-        poem_year = re.findall('[0-9]{4}', poem_year)
-
-        res.append([poem_name, poem_year[0], poem_ur])
+            poem_year = re.findall('[0-9]{4}', poem_year)
+            poem_year = poem_year[0]
+        except (AttributeError, IndexError):
+            poem_year = ''
+        res.append([poem_name, poem_year, poem_url])
     return res
 
 
-# парсить посилання на вірш
 def parse_poem_by_url(url):
+    """
+    Html parser for parsing selected poems.Returns poem name and whole poem text
+    :param url: poem url
+    :return: whole poem text
+    """
     poem_page = get_resource(url)
     soup = BeautifulSoup(poem_page, 'html.parser')
-    poem_header = soup.find('span', class_="text-head2").text
+    poem_header = soup.find('header', class_="entry-header clearfix").text
     poem_text = soup.find('p', align='left').text
-    poem = (poem_header + '\n' + poem_text + '\n')
+    poem = (f'{poem_header.strip()} ' + '\n'
+            f'{poem_text}' + '\n')
     return poem
 
 
 def print_whole_poem(poem_url):
+    """
+    :param poem_url: url of selected poem
+    :return: whole poem text
+    """
     poem = parse_poem_by_url(poem_url)
     print(poem)
+    return poem
+
+
+def parse_args():
+    """
+    :return:  command line arguments
+    """
+    parser = argparse.ArgumentParser(description='Processing of command line arguments')
+    parser.add_argument('--output_file', type=str)
+    parser.add_argument('--sort_poems', choices=['name', 'years'], nargs=1, type=str)
+    parser.add_argument('--sort_order', choices=['asc', 'desc'], default='asc', nargs=1, type=str)
+    parser.add_argument('--display_novel', type=lambda x: x.split('|'))
+    args = parser.parse_args()
+    return args
+
+
+def write_poem(poem, output_file):
+    """
+    Writes poem list in file and prints count of poem and absolute path to the  file.
+    :param poem: list of poems
+    :param output_file: command line argument
+    """
+    # create folder
+    path = os.path.dirname(os.path.abspath(output_file))
+    if not os.path.exists(path):
+        os.makedirs(path)
+    # open file
+    with open(output_file, "w", encoding='utf-8') as f:
+        for p in poem:
+            f.write(f'{p[0].strip()} {p[1].strip()} \n')
+    print('Count of poem =', len(poem))
+    print("Absolute path = ", path)
+
+
+def display_poem(poem, display_novel):
+    """
+    :param poem: list of poems
+    :param display_novel: command line argument
+    """
+    for p in poem:
+        for n in display_novel:
+            if n.strip() == p[0]:
+                return [[p[0], print_whole_poem(p[2])]]
```

Comment:
1. You have nested for statements, in this case it's possible to avoid this, you should use right structure, read more about data structures. https://docs.python.org/3/tutorial/datastructures.html
Take a look to the dictionaries: https://pythonworld.ru/tipy-dannyx-v-python/slovari-dict-funkcii-i-metody-slovarej.html